Mathlib/Algebra/Module/FinitePresentation.lean

```diff
@@ -73,21 +74,33 @@ end Semiring
 
 section Ring
 
-variable (R M N) [Ring R] [AddCommGroup M] [Module R M] [AddCommGroup N] [Module R N]
+section
+
+universe u v
+variable (R : Type u) (M : Type*) [Ring R] [AddCommGroup M] [Module R M]
+
+theorem Module.FinitePresentation.exists_fin [fp : Module.FinitePresentation R M] :
+    ∃ (n : ℕ) (K : Submodule R (Fin n → R)) (_ : M ≃ₗ[R] (Fin n → R) ⧸ K), K.FG := by
```

Comment:
Added this in analogy to [Module.Finite.exists_fin'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Finiteness/Cardinality.html#Module.Finite.exists_fin'), and refactored `.equiv_quotient`.